src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -165,6 +166,10 @@ private static IOperation CreateInternal(BoundNode boundNode)
                     return CreateBoundLabeledStatementOperation((BoundLabeledStatement)boundNode);
                 case BoundKind.ExpressionStatement:
                     return CreateBoundExpressionStatementOperation((BoundExpressionStatement)boundNode);
+                case BoundKind.InterpolatedString:
+                    return CreateBoundInterpolatedStringExpressionOperation((BoundInterpolatedString)boundNode);
+                case BoundKind.StringInsert:
+                    throw ExceptionUtilities.UnexpectedValue(boundNode.Kind);
```

Comment:
Binder only generates a BoundStringInsert node within BindInterpolatedString, and we handle  BoundKind.StringInsert in the newly added method below. Basically this validates the binder contract, we can have BoundStringInsert in the bind tree only under a BoundInterpolatedStringExpression.